bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -86,7 +86,41 @@ PrePrepareMsg::PrePrepareMsg(ReplicaId sender,
   } else {
     b()->digestOfRequests = nullDigest;
   }
+  b()->batch_cid_length = std::to_string(s).size();
+  b()->endLocationOfLastRequest = payloadShift();
+  b()->flags = computeFlagsForPrePrepareMsg(ready, ready, firstPath);
+  b()->numberOfRequests = 0;
+  b()->seqNum = s;
+  b()->viewNum = v;
 
+  char* position = body() + sizeof(Header);
+  memcpy(position, spanContext.data().data(), b()->header.spanContextSize);
+  position += b()->header.spanContextSize;
+  memcpy(position, std::to_string(s).data(), std::to_string(s).size());
+}
+
+PrePrepareMsg::PrePrepareMsg(ReplicaId sender,
+                             ViewNum v,
+                             SeqNum s,
+                             CommitPath firstPath,
+                             const concordUtils::SpanContext& spanContext,
+                             size_t size,
+                             const std::string& correlationID)
+    : MessageBase(sender,
+                  MsgCode::PrePrepare,
+                  spanContext.data().size(),
+                  (((size + sizeof(Header) + correlationID.size()) < maxMessageSize<PrePrepareMsg>())
+                       ? (size + sizeof(Header) + correlationID.size())
+                       : maxMessageSize<PrePrepareMsg>() - spanContext.data().size()))
```

Comment:
same here - talk to Dima

Reply:
@teoparvanov , I am fine with the change:)